crates/smelt-slurm/src/lib.rs

```diff
@@ -193,6 +195,37 @@ fn default_artifacts(working_dir: &Path) -> HashMap<String, String> {
             .to_string(),
     )])
 }
+
+/// Adds artifacts under /tmp/artifacts
+fn add_artifact_paths<'a>(
+    mut artifact_map: HashMap<String, String>,
+    artifacts_dir: &'a str,
+) -> Pin<Box<dyn Future<Output = std::io::Result<HashMap<String, String>>> + 'a>> {
+    Box::pin(async move {
+        // Create a recursive directory walker
+        let mut paths = tokio::fs::read_dir(artifacts_dir).await?;
+
+        // Process each entry in the directory
+        while let Some(entry) = paths.next_entry().await? {
+            let path = entry.path();
+
+            // Skip if it's a directory - we'll handle its contents separately
+            if path.is_dir() {
+                // Recursively process subdirectories
+                artifact_map = add_artifact_paths(artifact_map, path.to_str().unwrap()).await?;
+                continue;
+            }
+
+            // Convert path to string, skip if conversion fails
+            if let Some(path_str) = path.to_str() {
+                artifact_map.insert(path_str.to_string(), path_str.to_string());
+            }
+        }
+
+        Ok(artifact_map)
+    })
+}
+
```

Comment:
nit: making this an async function and passing in artifacts_dir as a `String` would be simpler, imo -- would remove the lifetime bound